logging.js

```diff
@@ -235,11 +238,23 @@ export class LoggingClient {
 	}
 
 	_throttle(log) {
+
+		const now = Date.now();
+
+		// rate limit number of errors to 100 per minute
+		while (this._logHistory.length > 0 && this._logHistory[0] < (now - MAXIMUM_LOGS_TIME_SPAN)) {
+			this._logHistory.shift();
+		}
+		if (this._logHistory.length >= MAXIMUM_LOGS_PER_TIME_SPAN) {
+			console.warn(`Logging rate limit of ${MAXIMUM_LOGS_PER_TIME_SPAN} reached in timespan of ${MAXIMUM_LOGS_TIME_SPAN}ms`);
```

Comment:
I wasn't sure if I should put something in the console here or not... it's not like anyone's every going to see it. But it seemed fine enough.